build.gradle

```diff
@@ -14,15 +14,30 @@ ext {
   baseVersion = "1.2"
   snapshotVersion = true
   releaseCandidate = 0
-  variants = [2.4]
+  variants = [2.4, 2.5]
   variant = System.getProperty("variant") as BigDecimal ?: variants.last()
   if (variant == 2.4) {
-    groovyVersion = "2.4.13"
+    groovyVersion = "2.4.15"
     minGroovyVersion = "2.4.0"
-    maxGroovyVersion = snapshotVersion ? "3.9.99" : "2.9.99" // TODO lock these down once Groovy 3 gets close to RC stage
+    groovyDependency = "org.codehaus.groovy:groovy-all:${groovyVersion}"
+  } else if (variant == 2.5) {
+    groovyVersion = "2.5.0"
+    minGroovyVersion = "2.5.0"
+    groovyDependency = [
+      "org.codehaus.groovy:groovy:${groovyVersion}",
+      "org.codehaus.groovy:groovy-groovysh:${groovyVersion}",
+      "org.codehaus.groovy:groovy-json:${groovyVersion}",
+      "org.codehaus.groovy:groovy-nio:${groovyVersion}",
+      "org.codehaus.groovy:groovy-macro:${groovyVersion}",
+      "org.codehaus.groovy:groovy-templates:${groovyVersion}",
+      "org.codehaus.groovy:groovy-test:${groovyVersion}",
+      "org.codehaus.groovy:groovy-sql:${groovyVersion}",
+      "org.codehaus.groovy:groovy-xml:${groovyVersion}",
+    ]
   } else {
     throw new InvalidUserDataException("Unknown variant: $variant. Choose one of: $variants")
   }
+  maxGroovyVersion = snapshotVersion ? "3.9.99" : "2.9.99" // TODO lock these down once Groovy 3 gets close to RC stage
```

Comment:
Where is `maxGroovyVersion` used?

Reply:
It is used for a runtime check by spock to fail fast if the available groovy version is not supported.